tests/ui/lint/dropping_references-can-fixed.stderr

```diff
@@ -11,108 +11,108 @@ note: the lint level is defined here
    |
 LL | #![deny(dropping_references)]
    |         ^^^^^^^^^^^^^^^^^^^
-help: use `let _ = ...` to ignore the expression or result
+help: remove the whole expression
    |
 LL -     drop(&SomeStruct);
-LL +     let _ = &SomeStruct;
+LL +
    |
 
 error: calls to `std::mem::drop` with a reference instead of an owned value does nothing
-  --> $DIR/dropping_references-can-fixed.rs:12:5
+  --> $DIR/dropping_references-can-fixed.rs:13:5
    |
 LL |     drop(&owned1);
    |     ^^^^^-------^
    |          |
    |          argument has type `&SomeStruct`
    |
-help: use `let _ = ...` to ignore the expression or result
+help: remove the whole expression
    |
 LL -     drop(&owned1);
-LL +     let _ = &owned1;
```

Comment:
Is this the best suggestion? Perhaps changing it to `drop(owned1)` is what the author really intended?

Reply:
I agree that suggesting `let _ = ...` is a bit silly when the `drop` argument has an explicit `&`. It's more reasonable for something like `drop(ref);`.

Reply:
> I agree that suggesting `let _ = ...` is a bit silly when the `drop` argument has an explicit `&`. It's more reasonable for something like `drop(ref);`.

`let _ = …` makes sense as a replacement when it actually drops something, e.g. for `let _ = func_call()`. `let _ = variable_containing_mut_ref;` is not useful as a replacement because it does exactly nothing, it simply matches the always matching pattern against the variable. The variable is not made inaccessible.

Reply:
> The variable is not made inaccessible.

That was never the point. The point of the lint is to signal users that the `drop` call does not drop the underline value.

Whenever or not the variable after the suggestion is made inaccessible is IMO completely irrelevant to the issue that the lint is trying to point at.

> `let _ = …` makes sense as a replacement when it actually drops something

Or as a way to suppress to the "unused variable" lint, as was called out by T-lang in https://github.com/rust-lang/rust/pull/109732#issuecomment-1496414416.

Reply:
> > `let _ = …` makes sense as a replacement when it actually drops something
> 
> Or as a way to suppress to the "unused variable" lint, as was called out by T-lang in [#109732 (comment)](https://github.com/rust-lang/rust/pull/109732#issuecomment-1496414416).

Yes. I think this should be a conscious decision though. `let _ = variable;` signals to the reader that the programmer wanted to explicitly silence the unused variable lint. Perhaps we could mention that in the replacement hint? "If you just want to mark the variable as used, use `let _ = variable`?

Reply:
Wasn't this already done through the help text: "help: use `let _ = ...` to ignore the expression or result" ?

Reply:
I didn't understand the wording that way.

I'd go for something like "help: user `let _ = …` to mark the variables as used". "to ignore the expression or result [or variable]" doesn't tell me (personally) that this is used to suppress the unused variable lint.